src/org/zaproxy/zap/extension/ascanrulesAlpha/HttpOnlySite.java

```diff
@@ -0,0 +1,227 @@
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.MISC;
+	}
+	
+	@Override
+	public int getRisk() {
+		return Alert.RISK_MEDIUM;
+	}
+
+	@Override
+	public int getCweId() {
+		return 311; // CWE-311: Missing Encryption of Sensitive Data
+	}
+
+	@Override
+	public int getWascId() {
+		return 4; // WASC-04: Insufficient Transport Layer Protection
+	}
+
+	@Override
+	public void init() {
+
+	}
+
+	public void raiseAlert(HttpMessage newRequest, String message) {
+		String newUri = newRequest.getRequestHeader().getURI().toString();
+		bingo(getRisk(), //Risk
+				Alert.CONFIDENCE_MEDIUM, //Confidence/Reliability
+				getName(), //Name
+				getDescription(), //Description
+				getBaseMsg().getRequestHeader().getURI().toString(), //Original URI
+				null, //Param
+				"", //Attack
+				Constant.messages.getString(MESSAGE_PREFIX+"otherinfo", message, newUri), //OtherInfo 
+				getSolution(), //Solution
+				"", //Evidence
+				getCweId(), //CWE ID
+				getWascId(), //WASC ID
+				newRequest); //HTTPMessage
+	}
+
+	public URI constructURI(String redirect, HttpMessage newRequest) {
+		URI oldURI = newRequest.getRequestHeader().getURI();
+		try {
+			URI newURI = new URI(oldURI, redirect, true);
+			return newURI;
+		} catch (URIException err) {
+			try {
+				URI newURI = new URI(oldURI, redirect, false);
+				return newURI;
+			} catch (URIException ex) {
+				return null;
+			}
+		}
+	}
+
+	@Override
+	public void scan() {
+		
+		if (getBaseMsg().getRequestHeader().isSecure()) { //Base request is HTTPS
+			if (log.isDebugEnabled()){
+				log.debug ("The original request was HTTPS, so there is not much point in looking further.");
+			}
+			return;
+		}
+
+		HttpMessage newRequest = getNewMsg();
+		try{
+			String host = newRequest.getRequestHeader().getURI().getHost();
+			String path = newRequest.getRequestHeader().getURI().getPath();
+			newRequest.getRequestHeader().setURI(new URI("https",null, host, 443, path));
+		} catch (URIException e) {
+			log.error("Error creating HTTPS URL from HTTP URL:", e);
+			return;
+		}
+
+		if (isStop()) {
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+getName()+" Stopping.");
+			}
+			return;
+		}
+
+		try {
+			int count = 0;
+			while(count < 10) {
+				if (isStop()) {
+					if (log.isDebugEnabled()) {
+						log.debug("Scanner "+getName()+" Stopping.");
+					}
+					return;
+				}
+				sendAndReceive(newRequest, false);
+				int status = newRequest.getResponseHeader().getStatusCode();
+				if(HttpStatusCode.isRedirection(status)) {
+					String redirect = newRequest.getResponseHeader().getHeader(HttpResponseHeader.LOCATION);
+					if(redirect != null && !redirect.isEmpty()) {
```

Comment:
We should handle the case when there's no redirection by raising an alert saying that "there's no automatic redirection" (or something like that).